robot/web/web.go

```diff
@@ -701,8 +828,20 @@ func (w *wrappedStreamWithRC) RecvMsg(m any) error {
 	err := w.ServerStream.RecvMsg(m)
 
 	if w.requestKey.Load() == nil {
+		resource := getResourceName(m, w.apiMethod)
+		if resource != "" {
+			if ok := w.rc.incrInFlight(resource); !ok {
+				return &RequestLimitExceededError{
+					resource: resource,
+					limit:    w.rc.inFlightLimit,
+				}
+			}
+		}
 		requestKey := buildRCKey(m, w.apiMethod)
-		w.requestKey.Store(&requestKey)
+		w.requestKey.Store(&streamRequestKey{
```

Comment:
@dgottlieb [Benji's question][1] got me taking a closer look at this. Is it a problem that we just call `Load` here instead of `Swap` and then checking if we lost the race?

[1]: https://github.com/viamrobotics/rdk/pull/5133/files#r2207573136